rest_sensors.cpp

```diff
@@ -1751,6 +1751,56 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
                         rspItemState[QString("Error : unknown Window open setting for %1").arg(sensor->modelId())] = map[pi.key()];
                     }
                 }
+                else if (rid.suffix == RConfigSwingMode)
+                {
+                    if (map[pi.key()].type() == QVariant::String && map[pi.key()].toString().size() <= 19)
```

Comment:
19 ?

Reply:
No reason to enter anything longer than the longest string ;)

Reply:
So 19 is basically `strlen("three quarters open")` a few lines down?

To be frank, I'd just remove the `&& map[pi.key()].toString().size() <= 19` check.
It brings no extra safety and a reader (like me) might wonder what this magical 19  means :)
It further changes the error message based on the string length.

Reply:
If you insist :)